app/src/main/java/com/example/mymusic/Notifications/Notifications.kt

```diff
@@ -0,0 +1,185 @@
+package com.example.mymusic.Notifications
+
+import android.app.*
+import android.app.NotificationManager.IMPORTANCE_LOW
+import android.app.PendingIntent.FLAG_UPDATE_CURRENT
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.media.MediaMetadata.METADATA_KEY_ALBUM
+import android.media.MediaMetadata.METADATA_KEY_ARTIST
+import android.support.v4.media.MediaMetadataCompat
+import android.support.v4.media.session.MediaSessionCompat
+import android.support.v4.media.session.PlaybackStateCompat
+import androidx.annotation.IdRes
+import androidx.core.app.NotificationCompat
+import androidx.core.graphics.toColorInt
+import androidx.media.session.MediaButtonReceiver
+import androidx.palette.graphics.Palette
+import com.example.mymusic.R
+import com.example.mymusic.constants.Constants.ACTION_NEXT
+import com.example.mymusic.constants.Constants.ACTION_PLAY_PAUSE
+import com.example.mymusic.constants.Constants.ACTION_PREVIOUS
+import com.example.mymusic.extensions.isPlaying
+import com.example.mymusic.playback.server.MediaSessionConnection
+import com.example.mymusic.playback.server.MediaSessionConnectionImpl
+import com.example.mymusic.playback.server.MusicService
+import com.example.mymusic.ui.activities.MainActivity
+import com.example.mymusic.util.Utils.isOreo
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import java.lang.System.currentTimeMillis
+
+private const val CHANNEL_ID = "music_channel_01"
+private const val NOTIFICATION_ID = 888
+
+interface Notifications {
+
+    fun updateNotification(mediaSession: MediaSessionCompat)
+
+    fun buildNotification(mediaSession: MediaSessionCompat): Notification
+}
+
+class NotificationImpl(
+    private val context: Context,
+    private val notificationManager: NotificationManager
+) : Notifications {
+    private var postTime: Long = -1
+
+    override fun updateNotification(mediaSession: MediaSessionCompat) {
+        GlobalScope.launch {
+            notificationManager.notify(NOTIFICATION_ID, buildNotification(mediaSession))
+        }
```

Comment:
GlobalScope cực kỳ nguy hiểm nhé. 
```suggestion
        notificationManager.notify(NOTIFICATION_ID, buildNotification(mediaSession))
```